packages/js/src/generators/library/library.ts

```diff
@@ -262,17 +264,13 @@ function createFiles(tree: Tree, options: NormalizedSchema, filesDir: string) {
     tmpl: '',
     offsetFromRoot: offsetFromRoot(options.projectRoot),
     rootTsConfigPath: getRelativePathToRootTsConfig(tree, options.projectRoot),
-    buildable: options.buildable === true,
+    buildable: options.bundler && options.bundler !== 'none',
     hasUnitTestRunner: options.unitTestRunner !== 'none',
   });
 
-  if (options.compiler === 'swc') {
+  if (options.bundler === 'swc' || options.bundler === 'rollup') {
     addSwcDependencies(tree);
-    addSwcConfig(
-      tree,
-      options.projectRoot,
-      options.bundler === 'rollup' ? 'es6' : 'commonjs'
-    );
+    addSwcConfig(tree, options.projectRoot, 'es6');
```

Comment:
question: are we defaulting `js:lib` with `swc` to `es6`? This might introduce unexpected behavior for nodejs folks.

Reply:
The expected behavior would be:
- `nrwl/js:swc` -> `commonjs`
- `nrwl/rollup:rollup` + `compiler: 'swc'` -> `es6`

Reply:
That's fair, we can keep it as commonjs unless the bundler is `vite` or `rollup` (ie. they are likely to target web).

Reply:
Ok, I updated the logic to reflect that!